interactions/api/models/gw.py

```diff
@@ -211,13 +212,374 @@ def __init__(self, **kwargs):
+    ) -> "GuildMember":
+        """
+        Modifies the member of a guild.
+
+        :param nick?: The nickname of the member
+        :type nick: Optional[str]
+        :param roles?: A list of all role ids the member has
+        :type roles: Optional[List[int]]
+        :param mute?: whether the user is muted in voice channels
+        :type mute: Optional[bool]
+        :param deaf?: whether the user is deafened in voice channels
+        :type deaf: Optional[bool]
+        :param channel_id?: id of channel to move user to (if they are connected to voice)
+        :type channel_id: Optional[int]
+        :param communication_disabled_until?: when the user's timeout will expire and the user will be able to communicate in the guild again (up to 28 days in the future)
+        :type communication_disabled_until: Optional[datetime.isoformat]
+        :param reason?: The reason of the modifying
+        :type reason: Optional[str]
+        :return: The modified member object
+        :rtype: Member
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        payload = {}
+        if nick is not MISSING:
+            payload["nick"] = nick
+
+        if roles is not MISSING:
+            payload["roles"] = roles
+
+        if channel_id is not MISSING:
+            payload["channel_id"] = channel_id
+
+        if mute is not MISSING:
+            payload["mute"] = mute
+
+        if deaf is not MISSING:
+            payload["deaf"] = deaf
+
+        if communication_disabled_until is not MISSING:
+            payload["communication_disabled_until"] = communication_disabled_until
+
+        res = await self._client.modify_member(
+            user_id=int(self.user.id),
+            guild_id=int(self.guild_id),
+            payload=payload,
+            reason=reason,
+        )
+        return GuildMember(**res, _client=self._client, guild_id=self.guild_id)
+
+    async def add_to_thread(
+        self,
+        thread_id: int,
+    ) -> None:
+        """
+        Adds the member to a thread.
+
+        :param thread_id: The id of the thread to add the member to
+        :type thread_id: int
+        """
+        if not self._client:
+            raise AttributeError("HTTPClient not found!")
+        await self._client.add_member_to_thread(
+            user_id=int(self.user.id),
+            thread_id=thread_id,
+        )
+
 
 class GuildMembers(DictSerializerMixin):
     """
     A class object representing the gateway event ``GUILD_MEMBERS_CHUNK``.
 
     :ivar Snowflake guild_id: The guild ID of the event.
-    :ivar List[Member] members: The members of the event.
+    :ivar List[GuildMember] members: The members of the event.
     :ivar int chunk_index: The current chunk index of the event.
     :ivar int chunk_count: The total chunk count of the event.
     :ivar list not_found: A list of not found members in the event if an invalid request was made.
```

Comment:
Should this be a list of integers, or user objects, or.. what?

Reply:
I changed it to the `GuildMember` object above, because if it has a guild_id in the event it seemed more logical to use it, than to make a normal member out of it

Reply:
or what exactly do you mean?